src/test/java/com/wikia/webdriver/common/templates/NewTestTemplate.java

```diff
@@ -8,37 +14,46 @@
 import com.wikia.webdriver.common.driverprovider.NewDriverProvider;
 import com.wikia.webdriver.common.driverprovider.UseUnstablePageLoadStrategy;
 
-import org.testng.SkipException;
-import org.testng.annotations.AfterMethod;
-import org.testng.annotations.BeforeMethod;
-
-import java.lang.reflect.Method;
-
 public class NewTestTemplate extends NewTestTemplateCore {
 
-  @BeforeMethod(alwaysRun = true)
-  public void start(Method method, Object[] data) {
-    Configuration.clearCustomTestProperties();
-    if (method.isAnnotationPresent(Execute.class)) {
-      if (!"".equals(method.getAnnotation(Execute.class).onWikia())) {
-        Configuration.setTestValue("wikiName", method.getAnnotation(Execute.class).onWikia());
-      }
-      if (!"".equals(method.getAnnotation(Execute.class).disableFlash())) {
-        Configuration
-            .setTestValue("disableFlash", method.getAnnotation(Execute.class).disableFlash());
-      }
+  private void setTestProperty(String key, String value) {
+    if (!"".equals(value)) {
+      Configuration.setTestValue(key, value);
     }
-    prepareURLs();
+  }
 
+  /**
+   * Return false if test is excluded from running on current test environment
+   * @param method
+   * @return
+   */
+  private boolean isTestExcludedFromEnv(Method method){
     if (method.isAnnotationPresent(DontRun.class)) {
       String[] excludedEnv = method.getAnnotation(DontRun.class).env();
       for (int i = 0; i < excludedEnv.length; i++) {
```

Comment:
or even better, loop is not necessary if you search in array
http://stackoverflow.com/questions/1128723/how-can-i-test-if-an-array-contains-a-certain-value